mesonbuild/build.py

```diff
@@ -114,22 +113,16 @@ def __init__(self, environment):
         self.targets = OrderedDict()
         # Coredata holds the state. This is just here for convenience.
         self.compilers = environment.coredata.compilers
-        self.cross_compilers = environment.coredata.cross_compilers
-        self.global_args = {}
-        self.projects_args = {}
-        self.global_link_args = {}
-        self.projects_link_args = {}
-        self.cross_global_args = {}
-        self.cross_projects_args = {}
-        self.cross_global_link_args = {}
-        self.cross_projects_link_args = {}
+        self.global_args = PerMachine({}, {}, {}) # PerMachine[Dict[str,List[str]]
```

Comment:
mypy can actually understand these if you put them in the form `# type: PerMachine[Dict[str, List[str]]`, Although, `PerMachine` would need to be defined as a generic for that to work.